pan-domain-auth-core/src/main/scala/com/gu/pandomainauth/PanDomainAuthSettingsRefresher.scala

```diff
@@ -1,83 +1,61 @@
+  * @param settingsFileKey the name of the file that contains the private settings for the given domain
+  * @param s3Client the AWS S3 client that will be used to download the settings from the bucket
+  * @param scheduler optional scheduler that will be used to run the code that updates the bucket
   */
 class PanDomainAuthSettingsRefresher(
   val domain: String,
   val system: String,
-  bucketName: String,
-  actorSystem: ActorSystem,
-  awsCredentialsProvider: AWSCredentialsProvider,
-  awsRegion: Regions,
-  proxyConfiguration: Option[ProxyConfiguration] = None
+  val bucketName: String,
+  settingsFileKey: String,
+  val s3Client: AmazonS3,
+  scheduler: ScheduledExecutorService = Executors.newScheduledThreadPool(1)
 ) {
-  lazy val bucket = new S3Bucket(bucketName, awsCredentialsProvider, awsRegion, proxyConfiguration)
-
-  private lazy val settingsMap = bucket.readDomainSettings(domain)
-  private lazy val authSettings: Agent[PanDomainAuthSettings] = Agent(PanDomainAuthSettings(settingsMap))
-
-  private lazy val domainSettingsRefreshActor = actorSystem.actorOf(Props(classOf[DomainSettingsRefreshActor], domain, bucket, authSettings), "PanDomainAuthSettingsRefresher")
-
-  actorSystem.scheduler.scheduleOnce(1 minute, domainSettingsRefreshActor, Refresh)
-
-  def settings = authSettings.get()
-}
-
-class DomainSettingsRefreshActor(domain: String, bucket: S3Bucket, authSettings: Agent[PanDomainAuthSettings]) extends Actor {
+  private val logger = LoggerFactory.getLogger(this.getClass)
 
-  val frequency: FiniteDuration = 1 minute
-  val log = Logging(context.system, this)
+  // This is deliberately designed to throw an exception during construction if we cannot immediately read the settings
+  private val authSettings: AtomicReference[PanDomainAuthSettings] = new AtomicReference[PanDomainAuthSettings](loadSettings() match {
+    case Right(settings) => PanDomainAuthSettings(settings)
+    case Left(err) => throw Settings.errorToThrowable(err)
+  })
 
-  override def receive: Receive = {
-    case Refresh => {
-      try {
-        val settingsMap = bucket.readDomainSettings(domain)
+  scheduler.scheduleAtFixedRate(() => refresh(), 1, 1, TimeUnit.MINUTES)
 
-        val settings = PanDomainAuthSettings(settingsMap)
+  def settings: PanDomainAuthSettings = authSettings.get()
 
-        authSettings send settings
-        log.debug("reloaded settings for {}", domain)
-      } catch {
-        case e: Exception => log.error(e, "failed to refresh domain {} settings", domain)
-      }
-      reschedule
-    }
+  private def loadSettings(): Either[SettingsFailure, Map[String, String]] = {
+    Settings.fetchSettings(settingsFileKey, bucketName, s3Client).flatMap(Settings.extractSettings)
   }
 
-  override def postRestart(reason: Throwable) {
-    reschedule
-  }
+  private def refresh(): Unit = {
+    loadSettings() match {
+      case Right(settings) =>
+        logger.info(s"Updated pan-domain settings for $domain")
```

Comment:
Wording changed on these logging messages for consistency with the public settings refresher